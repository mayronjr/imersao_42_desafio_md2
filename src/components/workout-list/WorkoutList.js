import 'bootstrap/dist/css/bootstrap.min.css';

import WorkoutRow from '../workout-row/WorkoutRow.js';

function WorkoutList({list}) {
    if(list !== undefined){
        return (
            <div className="container" data-testid="counter">
                {list.map((item, _index) => {
                    return (<WorkoutRow item={{...item, id:_index}} key={_index} />)
                })}
            </div>
        );    
    }else{
        return null
    }
    
}

export default WorkoutList;