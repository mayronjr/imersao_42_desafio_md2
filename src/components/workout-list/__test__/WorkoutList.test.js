import React from 'react'
import ReactDOM from 'react-dom'
import WorkoutList from '../WorkoutList.js'

import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"

import renderer from 'react-test-renderer'

const list = [
    { tempo: 1, type: 'Run', date: new Date(2011, 10, 24) },
    { tempo: 3, type: 'Swimming', date: new Date(2011, 8, 24) }
]

afterEach(cleanup)

it('renders without crashing', ()=>{
    const div = document.createElement('div')
    ReactDOM.render(<WorkoutList/>, div)
})

it('renders WorkoutList Correctly', ()=>{
    const {getByTestId} = render(<WorkoutList list={list}/>)
    expect(getByTestId('row-0')).toHaveTextContent("1Run11/24/2011")
    expect(getByTestId('row-1')).toHaveTextContent("3Swimming9/24/2011")
})

it('matches snapshot', ()=>{
    const tree = renderer.create(<WorkoutList list={list}/>).toJSON()
    expect(tree).toMatchSnapshot()
})