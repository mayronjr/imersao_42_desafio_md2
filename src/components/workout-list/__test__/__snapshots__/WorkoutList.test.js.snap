// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="counter"
>
  <div
    className="row-0"
    data-testid="row-0"
  >
    <div
      className="col-sm-4"
    >
      1
    </div>
    <div
      className="col-sm-4"
    >
      Run
    </div>
    <div
      className="col-sm-4"
    >
      11/24/2011
    </div>
  </div>
  <div
    className="row-1"
    data-testid="row-1"
  >
    <div
      className="col-sm-4"
    >
      3
    </div>
    <div
      className="col-sm-4"
    >
      Swimming
    </div>
    <div
      className="col-sm-4"
    >
      9/24/2011
    </div>
  </div>
</div>
`;
